app/src/main/java/in/testpress/testpress/ui/AttemptQuestionsFragment.java

```diff
@@ -258,18 +307,30 @@ public void createCheckBoxView(final List<AttemptAnswer> attemptAnswers, Attempt
             option.setId(i);
             LayoutParams layoutParams = new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
             option.setLayoutParams(layoutParams);
-            Spanned html = Html.fromHtml(answer, new AnswerImageGetter(), null);
+            Spanned html = Html.fromHtml(answer, new AnswerImageGetter(i), null);
+            final int noOfImages = answerImagesUrl.get(i + "").size();
             option.setText(trim(html, 0, html.length()));
-            for(int j = 0; j < answerImagesUrl.size() ; j++) {
-                imageLoader.loadImage(answerImagesUrl.get(j), new SimpleImageLoadingListener() {
-                    @Override
-                    public void onLoadingComplete(String imageUri, View view, Bitmap loadedImage) {
-                        Drawable drawable = new BitmapDrawable(loadedImage);
-                        answerImages.put(imageUri, drawable);
-                        Spanned html = Html.fromHtml(answer, new AnswerImageSetter(), null);
-                        option.setText(trim(html, 0, html.length()));
-                    }
-                });
+            for(int j = 0; j < noOfImages ; j++) {
+                if (answerImages.containsKey(answerImagesUrl.get(i + "").get(j))) {
+                        html = Html.fromHtml(answer, new AnswerImageSetter(), null);
+                        option.setText(clickableText(html));
+                        option.setMovementMethod(LinkMovementMethod.getInstance());
+                } else {
+                    imageLoader.loadImage(answerImagesUrl.get(i + "").get(j), new SimpleImageLoadingListener() {
+                        @Override
+                        public void onLoadingComplete(String imageUri, View view, Bitmap loadedImage) {
+                            Drawable drawable = new BitmapDrawable(loadedImage);
+                            answerImages.put(imageUri, drawable);
+                            noOfImagesLoaded++;
```

Comment:
The onLoadingComplete is an async callback. Hence initializing noOfImagesLoaded back to zero doesn't make sense as other callbacks will run independently.